### ISSUE TITLE
Saving notifications settings for "New posts" is flaky

### ISSUE THREAD

**Issue by designsimply** [wordpress-mobile/WordPress-Android#7796]
### Expected behavior

Changing the "New posts" setting in Me > Notification Settings would stay saved.

### Actual behavior

The "New posts" setting does not always seem to save properly in the Me > Notification Settings section.

### Steps to reproduce the behavior

1. Go to Me > Notification Settings.
1. Tap on any followed blog.
1. Toggle the "New posts" setting.
1. Tap "OK".
1. Tap on the same followed blog again and notice the setting that was just changed did not take.

Video: [1m13s](https://cloudup.com/cS83IdQ_OJA)

##### Tested on Pixel, Android 8.1.0, WPAndroid alpha-110

**Comment by theck13** [comment]
Thanks for reporting this!  It's not completely clear in the video which blog you're tapping every time and whether you're tapping ***Cancel*** or ***OK*** in the dialog.  Can you provide a script for the video or long-press on items for a second or two while recording?  That will help see what was tapped and what behavior to mimic when reproducing the issue.

Toggling the same notification setting on/off repeatedly and quickly was discussed in the pull request ending [here](https://github.com/wordpress-mobile/WordPress-Android/pull/7614#issuecomment-383608968).  Here's a synopsis.

<blockquote>
@nbradbury: If I toggle any of these radios and then tap OK to close the dialog, the changes aren't saved. They don't appear to be saved unless I wait to close the dialog for a couple seconds (or several seconds on a slow connection).
<blockquote>
@theck13: That's an unfortunate side effect of the dialog interface. We don't want to save the settings as they are toggled since the user can cancel the dialog and they would expect nothing to have changed even if they toggled something. Also, we don't necessarily want to have a blocking interface like a progress dialog after the user taps the OK button and confirms the changes. Due to that, we have to hit the API after the dialog is closed, which can take some time on slow connections as you mentioned.
<blockquote>
@SylvesterWilmott: I am not really expecting users to rapidly open/change/accept this dialog. But I will track this to be sure.
</blockquote>
</blockquote>
</blockquote>

**Comment by designsimply** [comment]
Sure!

Tested again and this time I found an additional crash scenario. If you try to change the setting for a deleted blog (such as convenient-cuban.jurassic.ninja) the app crashes. Then I tried changing the setting for a Jetpack site (alittletestblog.com) and the setting didn't take.

Video: [2m35s](https://cloudup.com/cSl2oRggzmr)
<sup>Tested with Pixel, Android 8.1.0, WPAndroid alpha-110.</sup>

**Comment by theck13** [comment]
> Tested again and this time I found an additional crash scenario. If you try to change the setting for a deleted blog (such as convenient-cuban.jurassic.ninja) the app crashes.

I wasn't able to reproduce the crash on a deleted WordPress.com site.  When tapping the ***OK*** button on the ***Notifications*** dialog after changing a setting for the site, the ***Followed Sites*** section is updated and the deleted site is removed without crashing.

I couldn't get it to crash with a deleted jurassic.ninja site either.  When tapping the ***OK*** button on the ***Notifications*** dialog after changing a setting for the site, the app doesn't crash.

Is there anything special about `convenient-cuban.jurassic.ninja?

> Then I tried changing the setting for a Jetpack site (alittletestblog.com) and the setting didn't take.

I can't reproduce this on a self-hosted site with Jetpack.  If I change a notification setting for `structural-louse-152.jurassic.ninja`, the change is applied.  Are there any other details about `alittletestblog.com` which might help troubleshooting?

**Comment by designsimply** [comment]
> Is there anything special about `convenient-cuban.jurassic.ninja`?

🤔 I deleted it via Network Admin > Sites > Deactivate (which might be considered a special thing).

> Are there any other details about alittletestblog.com which might help troubleshooting?

It's a Jetpack site, but I see you were testing Jetpack sites as well. I don't think the issue is blog-specific. It happens for other blogs for me as well, which you can see in the previous video.

**Comment by theck13** [comment]
> I deleted it via Network Admin > Sites > Deactivate (which might be considered a special thing).

That helped!  I was able to reproduce the crash using that method to delete a site.  I pinpointed the issue and submitted a fix in FluxC [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/815).  Once that change is merged, I'll open another pull request in WordPress-Android and that will take care of the crash.

> It's a Jetpack site, but I see you were testing Jetpack sites as well. I don't think the issue is blog-specific. It happens for other blogs for me as well, which you can see in the previous video.

I believe this is a consequence of the implementation I mentioned (i.e. we have to hit the API after the dialog is closed, which can take some time).  That may cause the setting to appear unchanged when tapping the site after closing the ***Notifications*** dialog in a short amount of time.  We don't expect users to be repeatedly and quickly changing the setting for a single site, but we are tracking that behavior and will change the implementation if it is common.

**Comment by theck13** [comment]
The crash fix has been merged and will be included in the next release.  Is it okay to close this issue, @designsimply?

**Comment by mzorz** [comment]
As part of the WPAndroid release process, I'm closing this issue as it seems to be tackled by the associated PR (https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/815) which was already merged.

Feel free to re-open if the issue reappears in the future. 👍
